src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,85 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  todo: Todo,
+  closeDetails: () => void,
+};
+
+export const TodoModal: React.FC<Props> = ({ todo, closeDetails }) => {
+  const {
+    id,
+    userId,
+    completed,
+    title,
+  } = todo;
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [selectedUser, setSelectedUser] = useState<User>({
+    email: '',
+    id: 0,
+    name: '',
+    phone: '',
+  });
+
+  useEffect(() => {
+    getUser(userId)
+      .then(user => {
+        setSelectedUser(user);
+        setIsLoaded(true);
+      })
+      .catch(error => `${error}`);
+  }, [userId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!isLoaded ? (
         <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      )
+        : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                Todo #
+                {id}
+              </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => {
+                  closeDetails();
+                }}
```

Comment:
```suggestion
                onClick={closeDetails()}
```

Reply:
okay, but without calling that callback :)